### ISSUE TITLE
feature request: recoll search results integration

### ISSUE THREAD

**Issue by pharmville** [linuxmint/Cinnamon#2692]
Nemo search bar and cinnamon menu applet would be more powerful (and faster) if they could index the contents of files, and list results based on the index. Obviously the locations that are indexed should be configurable.

I propose using recoll to do the difficult work here. It is very modular (it uses "filters" written in bash to extract text from various files). I have tested the most recent version of recoll and it quickly and accurately indexes all the files I have tried (mostly ms office and pdf files). Seems bug free in my testing compared to the alternatives.

In summary it would be nice if the cinnamon menu and nemo searches could display results from searching the recoll index. This type of integration is the only thing I miss from windows: I could press the windows key and type my search term and the right document would pop up in the list, even if the term was in the body of the document and not the filename)

**Comment by pgrm** [comment]
:+1:

**Comment by SamuelMiller** [comment]
Being able to search easily and quickly the content of files by specific criteria, as in Windows 7, whether it be within the main menu or nemo would be very useful! Unfortunately, this basic and essential feature is missing by default in many linux distros.

**Comment by JosephMcc** [comment]
Many thanks for contributing to Cinnamon. Your suggestion was reviewed. Your job is done and we'll take it from here :)

For more information on our workflow and feature requests, read https://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/faq.html

**Comment by Letterus** [comment]
May I ask if this request was integrated eventually? There is a GNOME Shell Search Provider for Recoll. Maybe that could be used for integration with Cinnamon or Nemo? https://www.lesbonscomptes.com/recoll/pages/download.html#gssp
